experimental/packages/exporter-trace-otlp-proto/src/platform/browser/OTLPTraceExporter.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReadableSpan, SpanExporter } from '@opentelemetry/sdk-trace-base';
+import { getEnv, baggageUtils } from '@opentelemetry/core';
+import {
+  OTLPExporterConfigBase,
+  appendResourcePathToUrl,
+  appendRootPathToUrlIfNeeded,
+} from '@opentelemetry/otlp-exporter-base';
+import {
+  OTLPProtoExporterBrowserBase,
+  ServiceClientType,
+} from '@opentelemetry/otlp-proto-exporter-base';
+import {
+  createExportTraceServiceRequest,
+  IExportTraceServiceRequest,
+} from '@opentelemetry/otlp-transformer';
+
+const DEFAULT_COLLECTOR_RESOURCE_PATH = 'v1/traces';
+const DEFAULT_COLLECTOR_URL = `http://localhost:4318/${DEFAULT_COLLECTOR_RESOURCE_PATH}`;
+
+/**
+ * Collector Trace Exporter for Web
+ */
+export class OTLPTraceExporter
+  extends OTLPProtoExporterBrowserBase<ReadableSpan, IExportTraceServiceRequest>
+  implements SpanExporter
+{
+  constructor(config: OTLPExporterConfigBase = {}) {
+    super(config);
+    this._headers = Object.assign(
+      this._headers,
+      baggageUtils.parseKeyPairsIntoRecord(
+        getEnv().OTEL_EXPORTER_OTLP_TRACES_HEADERS
+      )
+    );
+  }
+  convert(spans: ReadableSpan[]): IExportTraceServiceRequest {
+    return createExportTraceServiceRequest(spans, true);
+  }
+
+  getDefaultUrl(config: OTLPExporterConfigBase): string {
+    return typeof config.url === 'string'
```

Comment:
This is a little more readable, performant (and minifiable) by rewriting as
```ts
let endpoint = config.url;
if (!endpoint || endpoint !== 'string') {
    let env = getEnv();
    let tracesEndpoint = env.OTEL_EXPORTER_OTLP_TRACES_ENDPOINT;
    if (tracesEndpoint) {
        endpoint = appendRootPathToUrlIfNeeded(tracesEndpoint);
    } else {
        let otlpEndpoint = env.OTEL_EXPORTER_OTLP_ENDPOINT;
        if (otlpEndpoint) {
            endpoint = appendResourcePathToUrl(otlpEndpoint, DEFAULT_COLLECTOR_RESOURCE_PATH);
        }
    }
}

return endpoint || DEFAULT_COLLECTOR_URL;
```

Reply:
@MSNev The function `getDefaultUrl` is written this way in multiple packages. I suggest we take this up in a separate PR since it's unrelated to this PR.

Reply:
Can we raise an issue for this then, otherwise it will not get done. This can include the suggestion I have above as the implementation.